apiserver/facades/agent/deployer/deployer.go

```diff
@@ -99,7 +100,7 @@ func NewDeployerAPI(
 		PasswordChanger:        common.NewPasswordChanger(st, getAuthFunc),
 		APIAddresser:           common.NewAPIAddresser(systemState, resources),
 		UnitsWatcher:           common.NewUnitsWatcher(st, resources, getCanWatch),
-		StatusSetter:           common.NewStatusSetter(st, getAuthFunc),
+		StatusSetter:           common.NewStatusSetter(st, getAuthFunc, clock.WallClock),
```

Comment:
We should pass in the clock, we should not use the wall clock directly here. It's fine in tests, but not here.